dallinger/experiment.py

```diff
@@ -548,12 +556,7 @@ def retrieve_data(self):
 
     def end_experiment(self):
         """Terminates a running experiment"""
-        # Debug runs synchronously
-        if self.exp_config.get('mode') != 'debug':
-            self.log("Waiting for experiment to complete.", "")
-            while self.experiment_completed() is False:
-                time.sleep(30)
-            HerokuApp(self.app_id).destroy()
+        HerokuApp(self.app_id).destroy()
```

Comment:
Since you'll un-indent the call to `end_experiment` above, you should add the `if self.exp_config.get('mode') != 'debug':` around this call.